### ISSUE TITLE
Double-call to `filter_queryset()`

### ISSUE THREAD

**Issue by jnm** [kobotoolbox/kpi#2576]
The question is: can we (easily?) access the results of DRF's first `filter_queryset()` call when we're later in `AssetViewSet.get_serializer_context()`?
<hr>
Possibly unrelated unexpected behavior: 

SearchFilter.filter_queryset seems to be called twice per assets GET request. I don't see a reason for it to be called more than once. 

To see this in action, 

1. add a print statement in SearchFilter.filter_queryset in kpi/kpi/filters.py
2. request `/api/v2/assets/?format=json&q=` once. 
3. see your print statement appear twice in the kpi logs

Haven't looked deeply into this yet. So, I don't know what else the server might be doing twice per request.

_Originally posted by @p2edwards in https://github.com/kobotoolbox/kpi/issues/2514#issuecomment-556333262_

**Comment by jnm** [comment]
The first call to `filter_queryset()` is done internally by DRF. The second call happens in our `AssetViewSet.get_serializer_context()`:
https://github.com/kobotoolbox/kpi/blob/11a5a0da1999171339af1348a923c6843330dd42/kpi/views/v2/asset.py#L223-L245

First call:
```
  /opt/venv/lib/python3.8/site-packages/rest_framework/viewsets.py(114)view()
    113             # And continue as usual
--> 114             return self.dispatch(request, *args, **kwargs)
    115 

  /opt/venv/lib/python3.8/site-packages/rest_framework/views.py(502)dispatch()
    501 
--> 502             response = handler(request, *args, **kwargs)
    503 

  /opt/venv/lib/python3.8/site-packages/rest_framework/mixins.py(38)list()
     37     def list(self, request, *args, **kwargs):
---> 38         queryset = self.filter_queryset(self.get_queryset())
     39 

  /opt/venv/lib/python3.8/site-packages/rest_framework/generics.py(150)filter_queryset()
    149         for backend in list(self.filter_backends):
--> 150             queryset = backend().filter_queryset(self.request, queryset, self)
    151         return queryset

> /srv/src/kpi/kpi/filters.py(132)filter_queryset()
    131         import ipdb; ipdb.set_trace()
--> 132         try:
    133             q = request.query_params['q']
```
Second call:
```
  /opt/venv/lib/python3.8/site-packages/rest_framework/viewsets.py(114)view()
    113             # And continue as usual
--> 114             return self.dispatch(request, *args, **kwargs)
    115 

  /opt/venv/lib/python3.8/site-packages/rest_framework/views.py(502)dispatch()
    501 
--> 502             response = handler(request, *args, **kwargs)
    503 

  /opt/venv/lib/python3.8/site-packages/rest_framework/mixins.py(42)list()
     41         if page is not None:
---> 42             serializer = self.get_serializer(page, many=True)
     43             return self.get_paginated_response(serializer.data)

  /opt/venv/lib/python3.8/site-packages/rest_framework/generics.py(109)get_serializer()
    108         serializer_class = self.get_serializer_class()
--> 109         kwargs['context'] = self.get_serializer_context()
    110         return serializer_class(*args, **kwargs)

  /srv/src/kpi/kpi/views/v2/asset.py(231)get_serializer_context()
    230             asset_content_type = ContentType.objects.get_for_model(Asset)
--> 231             asset_ids = self.filter_queryset(queryset).values_list('id').distinct()
    232             object_permissions = ObjectPermission.objects.filter(

  /opt/venv/lib/python3.8/site-packages/rest_framework/generics.py(150)filter_queryset()
    149         for backend in list(self.filter_backends):
--> 150             queryset = backend().filter_queryset(self.request, queryset, self)
    151         return queryset

> /srv/src/kpi/kpi/filters.py(132)filter_queryset()
    131         import ipdb; ipdb.set_trace()
--> 132         try:
    133             q = request.query_params['q']
```

**Comment by noliveleger** [comment]
May be fixed by #2811
